src/ledger.ts

```diff
@@ -809,17 +817,26 @@ export class LedgerExportPublicKey extends LedgerExportHDNode {
    */
   async run() {
     try {
-      const result = await super.run();
-      const publicKey = this.parsePublicKey((result || {}).publicKey);
-      if (this.includeXFP) {
-        let rootFingerprint = await this.getXfp();
-        return {
-          rootFingerprint,
-          publicKey,
-        };
-      }
+      if (await this.isLegacyApp()) {
+        const result = await super.run();
+        const publicKey = this.parsePublicKey((result || {}).publicKey);
+        if (this.includeXFP) {
+          let rootFingerprint = await this.getXfp();
+          return {
+            rootFingerprint,
+            publicKey,
+          };
+        }
 
-      return publicKey;
+        return publicKey;
+      } else {
+        const publicKey = await this.getV2PublicKey();
+        if (this.includeXFP) {
+          let rootFingerprint = await this.getXfp();
+          return { rootFingerprint, publicKey };
+        }
+        return publicKey;
+      }
```

Comment:
the above if returns so we don't need this else

Reply:
done! thanks. [ee17f8b](https://github.com/unchained-capital/unchained-wallets/pull/95/commits/ee17f8b2e99bd3740dbd81f6a5ed0ac8060bf6d3)